README.md

```diff
@@ -153,7 +153,59 @@ Michaelstonis. (2022, November 4). _Model-View-ViewModel - .NET._ Microsoft Lear
 Ramotion. (2023, May 1). _Understanding MVVM: Model-View-ViewModel Architecture explained._ Web Design, UI/UX, Branding, and App Development Blog. https://www.ramotion.com/blog/what-is-mvvm/ 
 
 
+## Functional Patterns vs. OOP Design Patterns in JavaScript
 
+### Introduction: 
+
+In traditional Object-Oriented Programming (OOP), design patterns are widely used to solve common problems by providing structured approaches to design and development. However, with the rise of Functional Programming (FP), there's a shift towards different paradigms and patterns. In this Section, we'll explore the contrast between OOP design patterns and functional patterns in JavaScript.
+
+### OOP Design Patterns
+OOP design patterns like Singleton, Factory, Observer, and others are well-established concepts in JavaScript development. They focus on class hierarchies, encapsulation, and inheritance to achieve modularity and extensibility.
+
+For example, the Singleton pattern ensures that a class has only one instance and provides a global point of access to it. In JavaScript:
+
+Please see the example for [Singleton pattern in JavaScript](https://github.com/nic-dgl104-winter-2024/pattern-library/blob/main/patterns/Creational/singleton/JavaScript/singleton.js)
+
+### Functional Patterns
+Functional programming emphasizes immutable data and pure functions without side effects. While traditional OOP design patterns may not directly translate, functional programming offers its own set of patterns to address similar concerns.
+
+One common pattern in FP is the Higher-Order Function pattern. Higher-order functions take other functions as arguments or return them, enabling composition and abstraction.
+
+```
+JavaScript
+// Example of map, a higher-order function
+
+const numbers = [1, 2, 3, 4, 5];
+// Using the map method, which is a higher-order function,
+// to apply the arrow function to each element of the numbers array.
+// The arrow function takes an argument 'x' and returns 'x * 2'.
+const doubled = numbers.map(x => x * 2);
+console.log(doubled); // [2, 4, 6, 8, 10]
+```
+
+
+### Contrast and Comparison
+**Immutability:** FP favors immutable data, reducing the risk of unintended state changes. OOP, while supporting immutability, often relies on mutable state within objects.
+
+**Composition vs. Inheritance:** Functional programming encourages composition over inheritance. Instead of relying on class hierarchies, functions are composed to achieve complex behaviors.
+
+**Pattern Matching:** FP languages like Haskell or Scala offer pattern matching, allowing elegant handling of different cases. This contrasts with traditional switch-case or if-else statements in OOP.
+
```

Comment:
I like the table below, however it makes the statements above redundant. Perhaps as an improvement you could swap out the descriptions above with actual definitions for each. I think in this case it's fine to reproduce (i.e. quote) definitions directly from a reputable source - as long as you make clear they're quotes. For example, you could use the following for pattern matching:

> **Pattern Matching**: "the act of checking a given sequence of tokens for the presence of the constituents of some pattern" ([Wikipedia -Pattern matching](https://en.wikipedia.org/wiki/Pattern_matching)).

For the purposes of keeping things simple, include no more than one or two sentences for each of immutability, composition, and inheritance (you can use pattern matching as I've provided). Don't worry about explaining complicated terms, but do make sure you get the right definitions (i.e. object composition in programming - not musical composition!) And you can use Wikipedia as a source.

Reply:
Changes done

Reply:
I actually meant the other way around. I think the table was nice because it distinguished between functional and oop in an easy-to-read way. Do you mind updating one more time?:

- [ ] Definitions moved above the table and _replacing_ lines 187 - 191.
- [ ] Put the table back as you had it in the previous commit.
- [ ] I would also separate the definitions of composition and inheritance for clarity. The quote you sourced does explain the distinction, but it does really say what each one is. Wikipedia is a good source for both.

Reply:
Changes done.